MinecraftClient/Mapping/Entity.cs

```diff
@@ -48,12 +51,27 @@ public Entity(int ID, EntityType type, Location location)
         /// <param name="ID">Entity ID</param>
         /// <param name="type">Entity Type Enum</param>
         /// <param name="location">Entity location</param>
+        /// <param name="uuid">Player uuid</param>
         public Entity(int ID, EntityType type, Location location, Guid uuid)
         {
             this.ID = ID;
             this.Type = type;
             this.Location = location;
             this.UUID = uuid;
+            this.Name = "";
+            if (type == EntityType.Player)
+            {
+                
+                Dictionary<string, string> uuids = Program.Client.GetOnlinePlayersWithUUID();
```

Comment:
Maybe you can move this process into McTcpClient.OnEntitySpawn, this way no need to change visibility of Program.Client?

Reply:
Oh, also, the nickname might be present in entity NBT